core/src/main/scala/core/database/QueryRunner.scala

```diff
@@ -4,26 +4,36 @@ import javax.sql.DataSource
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class QueryRunner(
-    db: Database,
-    ec: ExecutionContext
-) {
-  def run[A](query: Query[A]): Future[A] =
+/**
+  * A class who can run a Query.
+  */
+sealed trait QueryRunner {
+  val ds: DataSource
+
+  val ec: ExecutionContext
```

Comment:
Even if each implementation of runner will use an `ExecutionContext`, I don't see the purpose of having it at this level of the API.